outlines/generate/generator.py

```diff
@@ -125,6 +123,12 @@ def token_generator(model, sampler: "Sampler") -> Callable:
 
     """
 
+    def cache_key_arguments(token_ids, attention_masks, kv_cache, allowed_tokens, rng):
```

Comment:
We should probably implement this function out of `token_generator` to keep that code simple.

Reply:
I'm not sure how we could do that, since we need to have access to model and sampler objects.